### PR TITLE
#137 added image preview

### LINKED ISSUE #137: As a user, I want an enhanced create a post page
### Title

As a user, I want an enhanced create a post page because I want to be able to see more information about what I'm posting.

### Describe the feature or problem you'd like to solve

A user should be able to see a preview of their uploaded image before submitting
The create a post page layout should be updated to make better use of the space (better design/layout)

### Dev notes

**User Interface**
image

- Do not implement a character counter for the caption
- Rename `Create a Post` button to `Upload` and place it under the Caption text field
- Rename HTML head title: Add a Picture -> Create a Post
- Limit the selectable file types in the file explorer pop up to the following
    - `.png`
    - `.jpg`
    - `.jpeg`
    - `.gif`

### Acceptance test
1. Go to the website
2. Log in with the following credentials: 
    - username: user1@test.com
    - password: !Test1
3. Open the Create a Post page
4. Upload an image and write a caption in the textarea

Expected result:
- A preview of the uploaded image will be displayed next to the caption textarea
- Post ready for submission

### PR DESCRIPTION
**Pull request by lealakkis**
## Screenshots
image
image

close #137

### CODE REVIEW

**Comment by lealakkis** [comment]
i had a hard time getting the image to show beside the description i tried a bunch of things so instead i moved it over and the image displays to the left if someone could take a look that would be appreciated.

**Review by jason-gerard — DISMISSED** [review]

File: app/forms/post_form.py [hunk 1]
```diff
@@ -8,12 +8,10 @@
 
 class CreatePostForm(FlaskForm):
     DESCRIPTION_LENGTH_MAX = 2_200
-    ALLOWED_EXTENSIONS = set([ 'png', 'jpg', 'jpeg', 'gif'])
-
 
     post_image = FileField(
         'Image',
-        render_kw={'style': 'height: auto', 'onchange':'previewImage(event)', 'accept': 'image/*'},
```
Comment (domsec) [inline app/forms/post_form.py]: Don't we want to limit the type of files displayed in the pop up file explorer to only our accepted image types?

Comment (jason-gerard) [reply app/forms/post_form.py]: sorry it was me who deleted that, ill add it back

**Comment by domsec** [comment]
Few more small refactors I'd suggest
- Navbar (Mobile): Create post -> Create a Post
- HTML head title: Add a Picture -> Create a Post
- Form button: Create a Post -> Upload
    - Having Create a Post be the same for the title and button feels redundant 
- Not sure if possible but maybe make the caption textarea unable to shrink less than the height of the left hand div (choose file + file preview)

**Review by davrine — APPROVED** [review]

**Review by davrine — DISMISSED** [review]

**Review by domsec — CHANGES_REQUESTED** [review]
View previous comments.

**Review by jason-gerard — DISMISSED** [review]

File: app/forms/post_form.py (hunk 1 above)
Comment (domsec) [reply app/forms/post_form.py]: Still not fixed

Comment (jason-gerard) [reply app/forms/post_form.py]: fixed

**Review by jason-gerard — APPROVED** [review]

**Review by domsec — APPROVED** [review]

**Pull request merged**